rust/agents/relayer/src/msg/pending_message.rs

```diff
@@ -331,6 +326,46 @@ impl PendingOperation for PendingMessage {
         }
     }
 
+    fn set_operation_outcome(
+        &mut self,
+        submission_outcome: TxOutcome,
+        submission_estimated_cost: U256,
+    ) {
+        let Some(operation_estimate) = self.get_tx_cost_estimate() else {
+            warn!("Cannot set operation outcome without a cost estimate set previously");
+            return;
+        };
+        if let Err(e) = self
+            .ctx
+            .origin_gas_payment_enforcer
+            .record_tx_outcome(&self.message, submission_outcome.clone())
+        {
+            error!(error=?e, "Error when recording tx outcome");
+        }
+        match gas_used_by_batch_operation(
```

Comment:
nit: maybe worth clarifying somewhere that this is used by non-batched operations too, took me a bit

Reply:
meant to rename this and forgot, thanks for pointing out